content/post/2018-03-30-sct-encoding.markdown

```diff
@@ -0,0 +1,413 @@
+
+In the decoded data, we can find the OID 1.3.6.1.4.1.11129.2.4.2, indicating
+the SCT list extension. Per [RFC 5280, section
+4.1](https://tools.ietf.org/html/rfc5280#page-17), an extension is defined:
+
+```
+Extension  ::=  SEQUENCE  {
+      extnID      OBJECT IDENTIFIER,
+      critical    BOOLEAN DEFAULT FALSE,
+      extnValue   OCTET STRING
+                  -- contains the DER encoding of an ASN.1 value
+                  -- corresponding to the extension type identified
+                  -- by extnID
+      }
+```
+
+We've found the "extnID". The "critical" field is omitted because it has the
+default value (false). Next up is the extnValue:
+
+This has the type OCTET STRING, which has the tag "0x04". It means "here's
+a bunch of bytes!" In this case, as described by the spec, those bytes
+happen to contain more DER. This is a fairly common pattern in X.509
+to deal with parameterized data. For instance, this allows defining a
+structure for extensions without knowing ahead of time all the structures
+that a future extension might want to carry in its value. Think of it as a
+`void*` (or `interface{}`) for data structures.
+
+Here's that extnValue:
+
+```
+0481F50481F200F0007500DB74AFEECB29ECB1FECA3E716D2CE5B9AABB36F7847183C75D9D4F37B61FBF64000001627313EB19000004030046304402207E1FCD1E9A2BD2A50A0C81E713033A0762340DA8F91EF27A48B3817640159CD30220659FE9F1D880E2E8F6B325BE9F18956D17C6CA8A6F2B12CB0F55FB70F759A419007700293C519654C83965BAAA50FC5807D4B76FBF587A2972DCA4C30CF4E54547F478000001627313EB2A0000040300483046022100AB72F1E4D6223EF87FC68491C208D29D4D57EBF47588BB7544D32F9537E2CEC10221008AFFC40CC6C4E3B24578DADE4F815ECBCE2D57A579342119A1E65BC7E5E69CE2
+```
+
+That's tag "0x04", meaning OCTET STRING, followed by "0x81 0xF5", meaning
```

Comment:
Might want to reference that `0x04` is the first byte (I don't think you can using the ``` syntax but it could also be nice to bold that byte in the string above.)